SystemAudioPlayer/impl/AudioPlayer.cpp

```diff
@@ -90,27 +93,49 @@ void AudioPlayer::Init(SAPEventCallback *callback)
     SAPLOG_INFO("SAP: AudioPlayer Init\n");
     if(!gst_is_initialized())
         gst_init(NULL,NULL);
+
     m_main_loop_thread = g_thread_new("BusWatch", (void* (*)(void*)) event_loop, NULL);
+    waitForLoopStart();
     m_callback = callback;
     Soc_Initialize();
 }
 
-void AudioPlayer::DeInit()
-{   
-    SAPLOG_INFO("SAP: AudioPlayer DeInit\n");
-    if(g_main_loop_is_running(m_main_loop))
-        g_main_loop_quit(m_main_loop);
-    g_thread_join(m_main_loop_thread);
-    Soc_Deinitialize();
-    SAPLOG_INFO("SAP: AudioPlayer DeInit last\n");
+void AudioPlayer::waitForLoopStart()
```

Comment:
A separate function might not be required, the logic can be simplified with std::condition_variable::wait_for() (this also handled spurious wake ups)

Reply:
done